Adds Popovers to all DatePicker examples and adjusts Popover behavior. (#1880)
iTwin/iTwinUI: apps/website/src/content/docs/datepicker.mdx

When a user needs to schedule deadlines, timeframes, or for age verification purposes, we provide them with a date picker. It behaves much like a select menu. We currently have three different types of date pickers.

## Usage

> The basic date picker allows the user to pick a date from the calendar and skip a month at a time.
> 
> This example and the others under this heading use the Popover utility component to show a real world use case.

Reviewer: Avoid terms like "this heading", which are too literal.

```suggestion
The `DatePicker` component allows the user to pick a date from the calendar and skip a month at a time.

This component is recommended to be used inside the [`Popover`](popover) component, which takes care for placement, collision handling, focus management, outside clicks, and other interaction details.
```

<LiveExample src='DatePicker.basic.jsx' truncate={true}>
  <AllExamples.DatePickerBasicExample client:load />
</LiveExample>

> If you would rather see an example of the date picker being used on its own, look at the <a href="#standalone-usage">standalone usage</a> example.

Reviewer: Prefer markdown syntax over HTML

```suggestion
If you would rather see an example of the date picker being used on its own, look at the [standalone usage](#standalone-usage).
```

> This allows the user to visually select the date rather than entering the date with a keyboard. This has the advantage of also showing weekdays, which may be important in a given context. The date picker is commonly used within column filters in [Table](table).

Reviewer: This one feels out-of-place with the new text added above. Maybe it's ok to remove, as it's a lot of fluff

```suggestion
```

  <AllExamples.DatePickerWithYearExample client:load />
</LiveExample>

> ## Standalone usage

Reviewer: Should be a sub-heading (h3) under "Usage" (h2)

```suggestion
### Standalone usage
```

> A date picker can be featured on your page in multiple ways. One can be added directly to your page so that it is visible as soon as the page loads. Additionally, a date picker can be added to a menu button on your page so that the user can toggle its visibility.

Reviewer: This text is out-of-date with the example. It should be focused on explaining the standalone usage, and specifically mention the `applyBackground` prop as well.

```suggestion
The `DatePicker` does not always need to be used within its own popover. You can instead display it as part of a larger interface containing interconnected controls.

You can set `applyBackground={false}` to make the `DatePicker` blend into its surroundings. This is useful when the surrounding UI already brings its own background-color and border.
```

Author: Added these in the latest push.

Reviewer: Missing second paragraph
